src/main/java/net/panda/mekanismfabric/MekanismDataGenerator.java

```diff
@@ -2,10 +2,224 @@
+		private static final String INGOT_TO_NUGGET_PATH = "ingot_to_nugget/";
+		private static final String BLOCK_TO_INGOT_PATH = "block_to_ingot/";
+		private static final String NUGGET_TO_INGOT_PATH = "nugget_to_ingot/";
+		private static final String ORE_TO_INGOT_PATH = "ore_to_ingot/";
+		private static final String DUST_TO_INGOT_PATH = "dust_to_ingot/";
+		private static final String SMELTING_PATH = "smelting/";
+		private static final String BLASTING_PATH = "blasting/";
+
+		private MekanismRecipeGenerator(FabricDataOutput output) {
+			super(output);
+		}
+
+		private String extractIdentifier(ItemConvertible output) {
+			String[] translationKeyParts = output.asItem().getTranslationKey().split("\\.");
+			return translationKeyParts[translationKeyParts.length - 1];
+		}
+
+		private void createShapelessRecipe(Consumer<RecipeJsonProvider> exporter, ItemConvertible input, ItemConvertible output, String path) {
+			ShapelessRecipeJsonBuilder.create(RecipeCategory.MISC, output, 9)
+					.input(input)
+					.criterion(FabricRecipeProvider.hasItem(output),
+							FabricRecipeProvider.conditionsFromItem(output))
+					.criterion(FabricRecipeProvider.hasItem(input),
+							FabricRecipeProvider.conditionsFromItem(input))
+					.offerTo(exporter, path + extractIdentifier(output));
+		}
+
+		private void create3x3ShapedRecipe(Consumer<RecipeJsonProvider> exporter, ItemConvertible input, ItemConvertible output, RecipeCategory category, String path) {
+			ShapedRecipeJsonBuilder.create(category, output)
+					.pattern("aaa").pattern("aaa").pattern("aaa")
+					.input('a', input)
+					.criterion(FabricRecipeProvider.hasItem(input),
+							FabricRecipeProvider.conditionsFromItem(input))
+					.criterion(FabricRecipeProvider.hasItem(output),
+							FabricRecipeProvider.conditionsFromItem(output))
+					.offerTo(exporter, path + extractIdentifier(output));
+		}
+
+		private void createShapelessBlockToIngotRecipe(Consumer<RecipeJsonProvider> exporter, ItemConvertible input, ItemConvertible output) {
+			createShapelessRecipe(exporter, input, output, BLOCK_TO_INGOT_PATH);
+		}
+
+		private void createShapelessIngotToNuggetRecipe(Consumer<RecipeJsonProvider> exporter, ItemConvertible input, ItemConvertible output) {
+			createShapelessRecipe(exporter, input, output, INGOT_TO_NUGGET_PATH);
+		}
+
+		private void createShapedIngotToBlockRecipe(Consumer<RecipeJsonProvider> exporter, ItemConvertible input, ItemConvertible output) {
+			create3x3ShapedRecipe(exporter, input, output, RecipeCategory.BUILDING_BLOCKS, INGOT_TO_BLOCK_PATH);
+		}
+
+		private void createShapedNuggetToIngotRecipe(Consumer<RecipeJsonProvider> exporter, ItemConvertible input, ItemConvertible output) {
+			create3x3ShapedRecipe(exporter, input, output, RecipeCategory.MISC, NUGGET_TO_INGOT_PATH);
+		}
+
+		private void createShapelessResourceBlockToIngotRecipe(Consumer<RecipeJsonProvider> exporter, PrimaryResources resource) {
```

Comment:
Rename `createShapelessResourceBlockToIngotRecipe` to `createShapelessBlockToIngotRecipeFromResource`